archive/android_bootldr_qcom.ksy

```diff
@@ -0,0 +1,179 @@
+  [foreign-others-shamu]: https://androidfilehost.com/?w=search&s=bootloader-shamu&type=files
+
+  [foreign-sample-angler]: https://androidfilehost.com/?fid=11410963190603870158 "bootloader-angler-angler-03.84.img"
+  [foreign-others-angler]: https://androidfilehost.com/?w=search&s=bootloader-angler&type=files
+
+  ---
+
+  The `bootloader-*.img` samples referenced above originally come from factory
+  images packed in ZIP archives that can be found on the page [Factory Images
+  for Nexus and Pixel Devices](https://developers.google.com/android/images) on
+  the Google Developers site. Note that the codenames on that page may be
+  different than the ones that are written in the list above. That's because the
+  Google page indicates **ROM codenames** in headings (e.g. "occam" for Nexus 4)
+  but the above list uses **model codenames** (e.g. "mako" for Nexus 4) because
+  that is how the original `bootloader-*.img` files are identified. For most
+  devices, however, these code names are the same.
+
+doc-ref: https://android.googlesource.com/device/lge/hammerhead/+/7618a7d/releasetools.py
+seq:
+  - id: magic
+    contents: BOOTLDR!
+  - id: num_images
+    type: u4
+  - id: ofs_img_bodies
+    -orig-id: start_offset
+    type: u4
+  - id: bootloader_size
+    -orig-id: bootldr_size
+    type: u4
+    doc: |
+      According to all available `releasetools.py` versions from AOSP (links are
+      in the top-level `/doc`), this should determine only the size of
+      `img_bodies` - there is [an assertion](
+      https://android.googlesource.com/device/lge/hammerhead/+/7618a7d/releasetools.py#167)
+      for it.
+
+      However, files for certain Pixel devices (see `/doc`) apparently declare
+      the entire file size here (i.e. including also fields from `magic` to
+      `img_headers`). So if you interpreted `bootloader_size` as the size of
+      `img_bodies` substream in these files, you would exceed the end of file.
+      Although you could check that it fits in the file before attempting to
+      create a substream of that size, you wouldn't know if it's meant to
+      specify the size of just `img_bodies` or the size of the entire bootloader
+      payload (whereas there may be additional data after the end of payload)
+      until parsing `img_bodies` (or at least summing sizes from `img_headers`,
+      but that's stupid).
+
+      So this field isn't reliable enough to be used as the size of any
+      substream. If you want to check if it has a reasonable value, do so in
+      your application code.
+  - id: img_headers
+    -orig-id: img_info
+    type: img_header
+    repeat: expr
+    repeat-expr: num_images
+instances:
+  img_bodies:
+    pos: ofs_img_bodies
+    type: img_body(_index)
+    repeat: expr
+    repeat-expr: num_images
+types:
+  img_header:
+    -webide-representation: '{name}'
+    seq:
+      - id: name
+        size: 64
+        type: strz
+        encoding: UTF-8
```

Comment:
@armijnhemel Doesn't

https://github.com/kaitai-io/kaitai_struct_formats/blob/82d9493383d9790eff5c3053f778d079b1509f00/archive/android_asus_bootldr.ksy#L14-L15

in fact apply here as well?